src/AutocompleteBadges.soy

```diff
@@ -0,0 +1,22 @@
+{namespace AutocompleteBadges}
+
+/**
+ * This renders the component's whole content.
+ * @param? badges
+ * @param? elementClasses
+ */
+{template .render}
+
+	<div class="autocomplete-badges {$elementClasses ?: ''}">
+		<textarea ref="input" autocomplete="off" class="autocomplete-field form-control" wrap="off"></textarea>
```

Comment:
Oh, and another thing I'd do is to use a text **input** tag instead of a **textarea** here, since it seems like you want it to be a single line of text anyway. It will remove that ugly scrollbar that shows up sometimes, making sure that the user can only type on a single line.